File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.cpp
class Solution {
public:
            string removeStars(string s) {
    int j = 0;
    for (int i = 0; i < s.size(); ++i)
        if (s[i] == '*')
            j--;
        else {
             //   cout<<"1 "<<j<<" "<<i<<endl;

            s[j] = s[i];
                j++;
                //cout<<"2 "<<j<<" "<<i<<endl;
        }
                    cout<<j;
    return s.substr(0,j);
    }
};
    /*    stack<char>st;
            for(int i=0;i<s.length();i++){
                    if(s[i]!='*'){
                            st.push(s[i]);
                    }else{
                            st.pop();
                    }
            }
            if(st.empty())
                    return "";
            string ss;
            while(!st.empty()){
                    ss+=st.top() ,st.pop();
            }
            reverse(ss.begin(),ss.end());
            return ss;
    }
};*/

File: 2390-removing-stars-from-a-string/2390-removing-stars-from-a-string.java
class Solution {
    public String removeStars(String s) {
          int j=0;
            char arr[]=s.toCharArray();
            for(int i=0;i<arr.length;i++){
                    if(arr[i]=='*')
                            j--;
                    else
                            arr[j++]=arr[i];
            }
            return String.valueOf(arr).substring(0,j);
    }
}
          /*  Stack<Character>st=new Stack<>();
                for(int i=0;i<s.length();i++)
                {
                  if(s.charAt(i)!='*'){
                        st.push(s.charAt(i));
                          continue;
                    }else
                        {
                          st.pop();
                  }                  
                }
            if(st.isEmpty())return "";
            
            StringBuilder sb=new StringBuilder();
            while(!st.isEmpty()){
                    sb.append(st.pop());
            }
            return sb.reverse().toString();
    }
}*/